### PR TITLE
Add NPM metadata to the create-plugin template

### PR DESCRIPTION
**Pull request by adamdmharvey**
## Hey, I just made a Pull Request!

Subsequent to #3585 and #3624, update the create-plugin template to properly add the NPM meta data.

Obviously, if a user were to want to use a plugin locally or seprately from the app at whole, they would modify this metadata, so I figured it was better to template it than not and have the users change it if they need to, than leave it empty.  Happy w/ alternate opinions on it. In future, could even have the cli ask a question about the URL if not defaulted, that sort of thing. For now, fired a patch to the cli for it.

image

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)

### CODE REVIEW

**Review by Rugvip — COMMENTED** [review]
The `create-plugin` command in the CLI is intended to be used in all Backstage repos, including internal ones created with `@backstage/create-app`. Not sure if that's what you mentioned in the PR body, but I think we should avoid adding this metadata to all created plugins.

Allowing this to be customized per repo could make sense, but not quite sure how to do that, open to suggestions :grin:

Maybe grab metadata from the root `package.json` and add it to the plugin one? Along with a modified `directory`.

**Comment by adamdmharvey** [comment]
Ah right, I see what you mean!  Meaning in my custom scaffolded app from say `npx @backstage/create-app`; got it.  Will take a peek!

**Comment by adamdmharvey** [comment]
As I noodle this out, I think for now I'll close this one and move on. 👍

File: packages/cli/templates/default-plugin/package.json.hbs [hunk 1]
```diff
@@ -13,6 +13,15 @@
     "main": "dist/index.esm.js",
     "types": "dist/index.d.ts"
   },
+  "homepage": "https://backstage.io",
```
Comment (andrewthauer) [inline packages/cli/templates/default-plugin/package.json.hbs:16]: I'm not sure this makes any sense in the template. Unless these can be customized via some template variables.

**Pull request closed without merging**